sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -1656,6 +1656,17 @@ object ExprChecks {
   }
 }
 
+object DriverTimeZone {
+
+  /**
+   * Check if the time zone of the driver is supported by plugin.
+   * @param driverZoneId time zone of the driver
+   */
+  def isSupportedByPlugin(driverZoneId: String): Boolean = {
+    driverZoneId == GpuOverrides.UTC_TIMEZONE_ID.toString
```

Comment:
I'd like to see this be part of `TypeChecks`, called something like `areTimestampsSupported`, and take a `ZoneId` which is normalized before checking against `UTC_TIMEZONE_ID`.  That way the code is more robust in handling timezones that are aliased.  As written, it will fail to compare them equal.

Regarding the comment of needing to remember to update the timestamp checking code, the best way to handle that is have the timezone checking code call this new method.  That way there's only one place that needs to be updated.

I see there's a number of places that are reaching into GpuOverrides to access this timezone ID and do their own checking, so those places should be updated to call this new TypeChecks method.  That can be done as a followup if desired.

Reply:
Added `areTimestampsSupported`  method in TypeChecks and updated the code in other places which were doing their own checks to use this method. PTAL.